hugr-core/src/import.rs

```diff
@@ -1022,32 +1022,38 @@ impl<'a> Context<'a> {
         &mut self,
         term_id: model::TermId,
     ) -> Result<ExtensionSet, ImportError> {
-        match self.get_term(term_id)? {
-            model::Term::Wildcard => Err(error_uninferred!("wildcard")),
+        let mut es = ExtensionSet::new();
+        let mut stack = vec![term_id];
 
-            model::Term::Var(var) => {
-                let mut es = ExtensionSet::new();
-                let (index, _) = self.resolve_local_ref(var)?;
-                es.insert_type_var(index);
-                Ok(es)
-            }
-
-            model::Term::ExtSet { extensions, rest } => {
-                let mut es = match rest {
-                    Some(rest) => self.import_extension_set(*rest)?,
-                    None => ExtensionSet::new(),
-                };
+        while let Some(term_id) = stack.pop() {
+            match self.get_term(term_id)? {
+                model::Term::Wildcard => return Err(error_uninferred!("wildcard")),
 
-                for ext in extensions.iter() {
-                    let ext_ident = IdentList::new(*ext)
-                        .map_err(|_| model::ModelError::MalformedName(ext.to_smolstr()))?;
-                    es.insert(&ext_ident);
+                model::Term::Var(var) => {
+                    let (index, _) = self.resolve_local_ref(var)?;
+                    es.insert_type_var(index);
                 }
 
-                Ok(es)
+                model::Term::ExtSet { items } => {
+                    for item in *items {
+                        match item {
+                            model::ExtSetItem::Extension(ext) => {
+                                let ext_ident = IdentList::new(*ext).map_err(|_| {
+                                    model::ModelError::MalformedName(ext.to_smolstr())
+                                })?;
+                                es.insert(&ext_ident);
+                            }
+                            model::ExtSetItem::Splice(term_id) => {
+                                stack.push(*term_id);
```

Comment:
Might be worth a comment that the final order doesn't matter (i.e. because it's an Extension_Set_)